src/helpers/radix_sort.rs

```diff
@@ -0,0 +1,247 @@
+
+    // Fold the result
+    let mut permutation_share = Vec::new();
+    (0..len).for_each(|j| {
+        permutation_share.push(mult_output[j] + mult_output[j + len]);
+    });
+    permutation_share
+}
+
+fn get_helper_specific_shares(
+    shares: &[ReplicatedShares],
+) -> Vec<Vec<ReplicatedFp31SecretSharing>> {
+    let mut helper_shares: Vec<Vec<ReplicatedFp31SecretSharing>> = Vec::new();
+    let mut helper1_share = Vec::new();
+    let mut helper2_share = Vec::new();
+    let mut helper3_share = Vec::new();
+
+    (0..shares.len()).for_each(|i| {
+        helper1_share.push(shares[i].0);
+        helper2_share.push(shares[i].1);
+        helper3_share.push(shares[i].2);
+    });
+
+    helper_shares.push(helper1_share);
+    helper_shares.push(helper2_share);
+    helper_shares.push(helper3_share);
+
+    helper_shares
+}
+
+#[cfg(test)]
+mod tests {
+
+    use crate::{
+        field::Fp31,
+        helpers::radix_sort::{helpers_do_multiplication, helpers_gen_permutations},
+        modulus_convert::{make_three, secret_share, ReplicatedFp31SecretSharing},
+    };
+
+    use super::{helpers_gen_mult_inputs, mpc_sort};
+
+    #[test]
+    fn replicated_share_sort() {
+        let input = [
+            (1, 0, 0),
+            (0, 0, 0),
+            (0, 1, 0),
+            (0, 0, 0),
+            (0, 0, 0),
+            (0, 0, 0),
+        ]
+        .to_vec();
+        let mut shares = Vec::new();
+        for iter in input {
+            shares.push(secret_share(iter.0, iter.1, iter.2));
+        }
+
+        let mpc_replicated_output = mpc_sort(&shares);
+        let expected_sort_output = [5_u128, 1, 6, 2, 3, 4].as_ref();
+
+        (0..mpc_replicated_output.len()).for_each(|i| {
+            assert_eq!(
+                mpc_replicated_output[i].0
+                    + mpc_replicated_output[i].1
+                    + mpc_replicated_output[i].2,
+                ReplicatedFp31SecretSharing::construct(
+                    Fp31::from(expected_sort_output[i]),
+                    Fp31::from(expected_sort_output[i])
+                )
+            );
+        });
+    }
+
+    #[test]
+    fn gen_mult_inputs() {
+        let helpers = make_three();
+        let input = [
+            (1, 0, 0),
+            (0, 0, 0),
+            (0, 1, 0),
+            (0, 0, 0),
+            (0, 0, 0),
+            (0, 0, 0),
+        ]
+        .to_vec();
+        let mut shares = Vec::new();
+        for iter in input {
+            shares.push(secret_share(iter.0, iter.1, iter.2));
+        }
+        let output = helpers_gen_mult_inputs(helpers, &shares);
+        let mult_output = output.1;
+
+        assert_eq!(3, mult_output.len());
+        assert_eq!(shares.len() * 2, mult_output[0].len());
+        assert_eq!(shares.len() * 2, mult_output[1].len());
+        assert_eq!(shares.len() * 2, mult_output[2].len());
+    }
+
+    #[test]
+    fn replicated_share_sort_test_individual_components_size_check() {
```

Comment:
Same as above, rather than having 3, highly repetitive tests, I'd recommend just adding these assets to the one main test.